01-syntax/03-data-structures/deep-comparation/deepEqual.js

```diff
@@ -3,16 +3,44 @@
  *
  */
 /* global deepEqual */
-
-// → Your code here.
-
 var obj = {
   here: {
     is: 'an'
   },
   object: 2
 };
+function deepEqual(object1, object2) {
+  var number1 = 0;
+  var number2 = 0;
+  var prop;
+  if (object1 === object2) {
+    return true;
+  }
+  function checkObject(object) {
+    var symbol = typeof boolean;
+    if (typeof object !== 'object' || object === null) {
+      symbol = false;
+    }
+    return symbol;
+  }
 
+  if (!checkObject(object1) || !checkObject(object2)) {
+    return false;
+  }
+  for (prop in object1) {
```

Comment:
Just an idea you can use [Object.keys](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/keys) for to count properties